testing/dart/canvas_test.dart

```diff
@@ -410,6 +410,109 @@ void main() {
     expect(areEqual, true);
   }, skip: !Platform.isLinux); // https://github.com/flutter/flutter/issues/53784
 
+  test('toGpuImage - too big', () async {
+    PictureRecorder recorder = PictureRecorder();
+    Canvas canvas = Canvas(recorder);
+    canvas.drawPaint(Paint()..color = const Color(0xFF123456));
+    final Picture picture = recorder.endRecording();
+    final Image image = picture.toGpuImage(300000, 4000000);
+    picture.dispose();
+
+    expect(image.width, 300000);
+    expect(image.height, 4000000);
+
+    recorder = PictureRecorder();
+    canvas = Canvas(recorder);
+
+    // On a slower CI machine, the raster thread may get behind the UI thread
+    // here. However, once the image is in an error state it will immediately
+    // throw on subsequent attempts.
+    while (true) {
```

Comment:
This makes me a bit uncomfortable, I'd be worried about the scenario where we break the error reporting somehow and this shard times out instead of fails.

If this had a max iteration count + perhaps a delay between drawImage attempts that might prevent that scenario, assuming there is no other way to stabilize it

Reply:
I don't want to introduce a delay - the await null is just to turn the event loop. I'm fine with having a max iteration count though.  I thought we had timeouts on these tests though - somewhere in litetest or something it should just eventually say the test failed due to timeout.

Reply:
Yeah as long as it times out before the CI shard times out, that is fine. The `await null` will run the next iteration in another microtask which not be enough, you need a non-zero duration await to get a full event loop

Reply:
TIL. I'll add a 1ms delay.